src/App.tsx

```diff
@@ -1,26 +1,258 @@
+  const [errorType, setErrorType] = useState<Error | null>(null);
+  const [filter, setFilter] = useState<Status>('all');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [focus, setFocus] = useState<boolean>(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [addNewTodo, setAddNewTodo] = useState<boolean>(false);
+  const [loadingAddTodoId, setLoadingAddTodoId] = useState<number | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [deleteFewTodo, setDeleteFewTodo] = useState<number[]>([]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+      } catch (err) {
+        setError(true);
+        setErrorType('load');
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    if (focus) {
+      const inputField = document.querySelector(
+        '.todoapp__new-todo',
+      ) as HTMLElement;
+
+      if (inputField) {
+        inputField.focus();
+      }
+    }
+  }, [focus]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleKeyDown = async (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    const trimmedTodo = newTodoTitle.trim();
+
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      if (trimmedTodo) {
+        setFocus(false);
+
+        try {
+          setAddNewTodo(true);
+
+          const newTodo: Todo = {
+            id: 0,
+            userId: USER_ID,
+            title: trimmedTodo,
+            completed: false,
+          };
+
+          setTempTodo(newTodo);
+
+          const response = await postTodo(newTodo);
+
+          setLoadingAddTodoId(response.id);
+
+          setTodos(prevTodos => [...prevTodos, response]);
+          setNewTodoTitle('');
+          setTempTodo(null);
+          setAddNewTodo(false);
+
+          setLoadingAddTodoId(null);
+          setFocus(true);
+        } catch (err) {
+          setError(true);
+          setErrorType('add');
+          setAddNewTodo(false);
+          setFocus(true);
+        }
+      } else {
+        setError(true);
+        setErrorType('empty');
+        setFocus(true);
+      }
+    }
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const onClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+    let errorAll = false;
+
+    for (const todo of completedTodos) {
+      try {
+        await deleteTodo(todo.id);
+      } catch (err) {
+        setError(true);
+        setErrorType('delete');
+        errorAll = true;
+      }
+    }
+
+    if (errorAll) {
+      setError(true);
+      setErrorType('delete');
+
+      return;
+    }
```

Comment:
if any of todos fail you have return and don't filter any of them in setTodos (thats why test fail)